### PR TITLE
add ppc/sh4 arches and linux guest fix

### PR DESCRIPTION
**Pull request by doomedraven**
* Arches
    * ppc
    * superH
* Fix linux guest issue
* update docs

### CODE REVIEW

File: cuckoo/core/guest.py [hunk 1]
```diff
@@ -337,7 +337,10 @@ def query_environ(self):
     def determine_analyzer_path(self):
         """Determine the path of the analyzer. Basically creating a temporary
         directory in the systemdrive, i.e., C:\\."""
-        systemdrive = "%s\\" % self.environ["SYSTEMDRIVE"]
+        if self.platform == "windows":
+            systemdrive = "%s\\" % self.environ["SYSTEMDRIVE"]
+        elif self.platform in ("linux", "darwin"):
+            systemdrive = self.environ["HOME"]
```
Comment (jbremer) [inline cuckoo/core/guest.py]: These changes surely require some unit tests.

File: cuckoo/core/guest.py [hunk 2]
```diff
@@ -460,9 +463,15 @@ def start_analysis(self, options, monitor):
 
         # If the target is a file, upload it to the guest.
         if options["category"] == "file" or options["category"] == "archive":
+
```
Comment (jbremer) [inline cuckoo/core/guest.py]: No newline please :)

File: cuckoo/core/guest.py [hunk 3]
```diff
@@ -460,9 +463,15 @@ def start_analysis(self, options, monitor):
...
+            if self.platform in ("linux", "darwin"):
+                temp = "/tmp"
```
Comment (jbremer) [inline cuckoo/core/guest.py]: Rename to `tempdir` or `dirpath`.

File: cuckoo/core/guest.py [hunk 4]
```diff
@@ -460,9 +463,15 @@ def start_analysis(self, options, monitor):
 
         # If the target is a file, upload it to the guest.
         if options["category"] == "file" or options["category"] == "archive":
+
+            if self.platform in ("linux", "darwin"):
```
Comment (jbremer) [inline cuckoo/core/guest.py]: I find `if self.platform == "linux" or self.platform == "darwin":` to be slightly neater in terms of code syntax.

File: cuckoo/data-private/cwd/conf/qemu.conf [hunk 5]
```diff
@@ -19,7 +19,7 @@ label = {{ qemu[machine].label }}
 # image path
 image = {{ qemu[machine].image }}
 
-# vm arch (mips/mipsel/arm/x64/x86)
+# vm arch (sh4/sparc/sparc64/powerpc/powerpc64/mips/mipsel/arm/x64/x86)
```
Comment (jbremer) [inline cuckoo/data-private/cwd/conf/qemu.conf]: I think x86/x64/arm take precedence over the others, so please sort them accordingly.

**Comment by doomedraven** [comment]
closing as integrated here https://github.com/cuckoosandbox/cuckoo/pull/1545

**Pull request closed without merging**